tests/app.test.js

```diff
@@ -355,6 +357,107 @@ describe('InVEST global settings: dialog interactions', () => {
     expect(input.classList.contains('is-invalid')).toBeTruthy();
     expect(getByText('Save Changes')).toBeDisabled();
   });
+
+  test('Save global settings', async () => {
+    const nWorkers = '2';
+    const loggingLevel = 'DEBUG';
+    const nWorkersLabelText = 'Taskgraph n_workers parameter';
+    const loggingLabelText = 'Logging threshold';
+
+    const { getByText, getByLabelText, getByTitle } = render(
+      <App investExe="foo" />
+    );
+
+    fireEvent.click(getByTitle('settings'));
+    const nWorkersInput = getByLabelText(nWorkersLabelText, { exact: false });
+    const loggingInput = getByLabelText(loggingLabelText, { exact: false });
+
+    // Test that the default values when no global-settings file exists are
+    // loaded. I've found this helps allow componentDidMount processes to
+    // finish
+    await waitFor(() => {
+      expect(nWorkersInput).toHaveValue("-1");
+      expect(loggingInput).toHaveValue("INFO");
+    });
+
+    // Change the value
+    fireEvent.change(nWorkersInput, { target: { value: nWorkers } });
+    fireEvent.change(loggingInput, { target: { value: loggingLevel } });
+    await waitFor(() => { // the value to test is inherited through props
```

Comment:
I think the change handler only updates the local state, right? No problem to use `waitFor` anyway, but I think the comment is inaccurate.